source/overloading.tex

```diff
@@ -3486,24 +3480,23 @@
 \indextext{overloading!unary operator}
 
 \pnum
-A prefix unary operator shall be implemented by a
-non-static member function\iref{class.mfct} with no parameters or a
-non-member function with one parameter.
+A \defnadj{prefix unary}{operator function} \tcode{\keyword{operator}@}
+for a prefix unary operator \tcode{@}\iref{expr.unary.op} is
+a non-static member function\iref{class.mfct} with no parameters or
+a non-member function with one parameter.
```

Comment:
I find this definition a bit confusing. I want to look at this and determine whether a particular function is a prefix unary operator function, and it doesn't seem to quite get us there (it reads like the choice of `@` is part of the input, not part of the definition). Maybe reorder a little:

> A *prefix unary operator function* is a function named `operator@` for a prefix unary operator `@` that is either a non-static member function with no parameters or a non-member function with one parameter.

or

> An operator function `operator@` is a *prefix unary operator function* if `@` is a prefix unary operator and the function is either a non-static member function with no parameters or a non-member function with one parameter.

or something.

Reply:
The first alternative sounds promising.

Reply:
Fixed.